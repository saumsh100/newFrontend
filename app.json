  {
  "name": "carecru",
  "scripts": {
  },
  "env": {
    "NODE_ENV": {
      "required": true
    },
    "RETHINKDB_AUTHKEY": {
      "required": true
    },
    "RETHINKDB_DB": {
      "required": true
    },
    "RETHINKDB_HOST": {
      "required": true
    },
    "RETHINKDB_PORT": {
      "required": true
    },
    "VENDASTA_API_KEY": {
      "required": true
    },
    "VENDASTA_API_USER": {
      "required": true
    },
    "TWILIO_ACCOUNT_SID": {
      "required": true
    },
    "TWILIO_AUTH_TOKEN": {
      "required": true
    },
    "TWILIO_NUMBER": {
      "required": true
    }
  },
  "formation": {
    "web": {
      "quantity": 1
    }
  },
  "addons": [

  ],
  "buildpacks": [
    {
      "url": "heroku/nodejs"
    }
  ]
}
