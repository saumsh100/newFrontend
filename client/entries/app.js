
import '../apps/default';
import '../apps/Dashboard';
