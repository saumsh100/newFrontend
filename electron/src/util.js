
module.exports.getChannelFromVersion = (version) => {
  const splitVersion = version.split('-');
  return splitVersion[1] || null;
};
