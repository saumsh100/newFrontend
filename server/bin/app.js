
require('../config/initializeCodeTranspiler');

const globals = require('../config/globals');
const applyWebpack = require('../config/webpack/applyWebpack');
const handleErrors = require('../middleware/handleErrors');
const chalk = require('chalk');
const app = require('../config/express');

// require('../config/kue');

// Uses the NODE_ENV to determine logging full stack or not
app.set('showStackError', true);

// Load in webpack configurations
applyWebpack(app);


// Set the Handlebars templating engine
app.set('views', `${globals.root}/views`);
app.engine('hbs', require('../config/handlebars').engine);
app.set('view engine', 'hbs');

// Extra logging for communication with server
app.use((req, res, next) => {
  console.log(chalk.blue(req.method, req.originalUrl));
  next();
});


// Bind the routes

app.use(require('../routes'));

// Catch errors, log and respond to client
app.use(handleErrors);

// assume 404 since no middleware responded
/*app.use((req, res) => {
  res.status(404).render('404', {
    url: req.originalUrl,
    error: 'Not found',
  });
});*/

module.exports = app;
