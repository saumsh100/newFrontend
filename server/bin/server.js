
const http = require('http');
const globals = require('../config/globals');
const app = require('./app');
const Availability = require('../models/Availability');
const Patient = require('../models/Patient');

const server = http.createServer(app);

const io = require('socket.io')(server);

// TODO: implement Thinky!!!
io.on('connection', (socket) => {
  console.log('a new socket connected');
  socket.on('fetchAvailabilities', () => {
    Availability.run()
      .then((availabilities) => {
        socket.emit('receiveAvailabilities', availabilities);
      });
  });
  
  socket.on('fetchPatients', () => {
    Patient.run()
      .then((patients) => {
        socket.emit('receivePatients', patients);
      });
  });
  
  socket.on('addAvailability', (data) => {
    new Availability(data).save().then((availability) => {
      io.sockets.emit('availabilityAdded', availability);
    });
  });
  
  socket.on('removeAvailability', (data) => {
    Availability.get(data.id).then((availability) => {
      availability.delete()
        .then((result) => {
          io.sockets.emit('availabilityRemoved', result);
        });
    });
  });
  
  socket.on('syncClientTrigger', (data) => {
    console.log('sync client triggers!', 'data:', data);
  });
});

server.listen(globals.port, () => {
  console.log(`CareCru HTTP Server is running on port ${globals.port}`);
});
