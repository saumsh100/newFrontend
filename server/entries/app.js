
import '../../client/apps/default';
import '../../client/apps/Dashboard';
