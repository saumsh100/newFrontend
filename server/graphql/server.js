
import { Router } from 'express';
import graphQLHTTP from 'express-graphql';
import { sequelizeAuthMiddleware } from '../middleware/auth';
import schema from './data/schema';

const GraphQLRouter = Router();

const PORT = 4000;

const GraphQLServer = graphQLHTTP({
  schema,
  graphiql: true,
  pretty: true,
  subscriptionsEndpoint: `ws://localhost:${PORT}/subscriptions`,
});

// Reusing auth middleware as the server is exposed the same way an API would.
GraphQLRouter.all('*', sequelizeAuthMiddleware, GraphQLServer);

export default GraphQLRouter;

