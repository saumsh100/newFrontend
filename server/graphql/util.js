
import { getOffsetWithDefault, offsetToCursor } from 'graphql-relay';

/**
 * A simple function that accepts an array and connection arguments, and returns
 * a connection object for use in GraphQL
 *
 * This is a fork of connectionFromArraySlice
 * from https://github.com/graphql/graphql-relay-js/blob/master/src/connection/arrayconnection.js#L65
 *
 * Diferente form the original, is this case we build the cursor and pageInfo based on args
 * whitout slicing the array.
 * That way we can pass the already sliced or filtered array to the function.
 *
 * For now, this only works with forward pagination.
 */
// eslint-disable-next-line import/prefer-default-export
export function connectionFromArrayWithoutSlice(array, args, meta) {
  const { after, first, last } = args;
  const { arrayLength } = meta;

  if (typeof first === 'number') {
    if (first < 0) {
      throw new Error('Argument "first" must be a non-negative integer');
    }
  }
  if (typeof last === 'number') {
    if (last < 0) {
      throw new Error('Argument "last" must be a non-negative integer');
    }
  }

  const afterOffset = getOffsetWithDefault(after, 0);

  const startOffset = Math.max(afterOffset, -1) + 1;
  const endOffset = startOffset + first;

  const edges = array.map((value, index) => ({
    cursor: offsetToCursor(startOffset + index),
    node: value,
  }));

  const firstEdge = edges[0];
  const lastEdge = edges[edges.length - 1];
  const lowerBound = after ? afterOffset + 1 : 0;
  const upperBound = arrayLength;

  return {
    edges,
    pageInfo: {
      startCursor: firstEdge ? firstEdge.cursor : null,
      endCursor: lastEdge ? lastEdge.cursor : null,
      hasPreviousPage: typeof last === 'number' ? startOffset > lowerBound : false,
      hasNextPage: typeof first === 'number' ? endOffset < upperBound : false,
    },
  };
}
