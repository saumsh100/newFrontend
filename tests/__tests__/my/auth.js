
import request from 'supertest';
import app from '../../../server/bin/app';
import { AuthSession, PatientUser, Patient, PinCode } from '../../../server/models';
import wipeModel, { wipeAllModels } from '../../util/wipeModel';
import { accountId, enterpriseId, seedTestUsers } from '../../util/seedTestUsers';
import { patientUser, patientUserId, seedTestPatients } from '../../util/seedTestPatients';
import { omitPropertiesFromBody, omitProperties } from '../../util/selectors';


const pinCodeId = '76d50acd-44f1-414e-bc9b-c0d197ce7149';
const authSessionId = 'c5574713-835d-4371-8cfe-8ad242a33499';

const pinCode = {
  id: pinCodeId,
  pinCode: '1234',
  modelId: patientUserId,
};

const authSession = {
  modelId: patientUserId,
  accountId,
  enterpriseId,
};

async function generatePatientUserToken() {
  const response = await request(app)
    .post('/auth')
    .set('Host', 'my.test.com')
    .send({
      email: patientUser.email,
      password: '!@CityOfBudaTest#$',
    });

  return response.body.token;
}

async function seedTestAuthSessions() {
  await wipeModel(AuthSession);
  await AuthSession.save(authSession);
}

async function seedTestPinCodes() {
  await wipeModel(PinCode);
  await PinCode.save(pinCode);
}

describe('/auth', () => {
  beforeAll(async() => {
    await seedTestUsers();
  });

  afterAll(async () => {
    await wipeAllModels();
  });

  describe('GET /', () => {
    beforeEach(async () => {
      await seedTestPatients();
      await seedTestAuthSessions();
    });

    test('/me - get current patientUser', async () => {
      const token = await generatePatientUserToken();
      return request(app)
        .get(`/auth/me?patientUserId=${patientUserId}&sessionId=${authSessionId}`)
        .set('Host', 'my.test.com')
        .set('Authorization', `Bearer ${token}`)
        .send(Object.assign(patientUser, { confirmPassword: patientUser.password }))
        .expect(200)
        .then(({ body }) => {
          body.patientUser = omitProperties(body.patientUser, ['password']);
          body = omitProperties(body, ['sessionId']);
          expect(body).toMatchSnapshot();
        });
    });
  });

  describe('POST /', () => {
    beforeEach(async () => {
      await wipeModel(PatientUser);
      await wipeModel(Patient);
      await wipeModel(PinCode);
    });

    test('/signup - sign up a patientUser', () => {
      return request(app)
        .post('/auth/signup')
        .set('Host', 'my.test.com')
        .send(Object.assign(patientUser, { confirmPassword: patientUser.password }))
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toHaveProperty('token');
        });
    });

    test('/signup/:patientUserId/confirm - confirm patientUser phoneNumber - successful', async () => {
      await seedTestPatients();
      await seedTestPinCodes();
      return request(app)
        .post(`/auth/signup/${patientUserId}/confirm`)
        .set('Host', 'my.test.com')
        .send({ confirmCode: pinCode.pinCode })
        .expect(200)
        .then(({ body }) => {
          body = omitProperties(body, ['password']);
          expect(body).toMatchSnapshot();
        });
    });

    test('/signup/:patientUserId/confirm - confirm patientUser phoneNumber - unsuccessful', async () => {
      await seedTestPatients();
      await seedTestPinCodes();
      return request(app)
        .post(`/auth/signup/${patientUserId}/confirm`)
        .set('Host', 'my.test.com')
        .send({ confirmCode: '4321' })
        .expect(400)
        .then(({ body }) => {
          body = omitProperties(body, ['password']);
          expect(body).toMatchSnapshot();
        });
    });

    // TODO: Figure out why async jasmine error occurs
    /*
    test.only('/:patientUserId/resend - resend sms to patient', async () => {
      await seedTestPatients();
      return request(app)
        .post(`/auth/${patientUserId}/resend`)
        .set('Host', 'my.test.com')
        .send(Object.assign(patientUser, { patientUserId }))
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });
    */

    test('/ - login - successful', async () => {
      await seedTestPatients();
      return request(app)
        .post('/auth')
        .set('Host', 'my.test.com')
        .send({
          email: patientUser.email,
          password: '!@CityOfBudaTest#$',
        })
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toHaveProperty('token');
        });
    });

    test('/ - login - unsuccessful', async () => {
      await seedTestPatients();
      return request(app)
        .post('/auth')
        .set('Host', 'my.test.com')
        .send({
          email: patientUser.email,
          password: 'wrongpassword',
        })
        .expect(401)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });
  });

  describe('DELETE /', () => {
    beforeEach(async () => {
      await seedTestPatients();
      await seedTestAuthSessions();
    });

    test('/session/:sessionId - log out patientUser', async () => {
      return request(app)
        .delete(`/auth/session/${authSessionId}`)
        .set('Host', 'my.test.com')
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });
  });
});
