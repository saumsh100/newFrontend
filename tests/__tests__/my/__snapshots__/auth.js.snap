// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/auth DELETE / /session/:sessionId - log out patientUser 1`] = `
Object {
  "entities": Object {},
}
`;

exports[`/auth GET / /me - get current patientUser 1`] = `
Object {
  "patientUser": Object {
    "avatarUrl": null,
    "email": "testpatientuser@test.com",
    "firstName": "Jack",
    "id": "b8665c97-1e98-41ac-bf13-090b742de400",
    "isEmailConfirmed": false,
    "isPhoneNumberConfirmed": false,
    "lastName": "Daniels",
    "patientUserId": null,
    "phoneNumber": "+16049999999",
  },
}
`;

exports[`/auth POST / / - login - unsuccessful 1`] = `
Object {
  "entities": Object {},
}
`;

exports[`/auth POST / /signup/:patientUserId/confirm - confirm patientUser phoneNumber - successful 1`] = `
Object {
  "avatarUrl": null,
  "email": "testpatientuser@test.com",
  "firstName": "Jack",
  "id": "b8665c97-1e98-41ac-bf13-090b742de400",
  "isEmailConfirmed": false,
  "isPhoneNumberConfirmed": true,
  "lastName": "Daniels",
  "patientUserId": null,
  "phoneNumber": "+16049999999",
}
`;

exports[`/auth POST / /signup/:patientUserId/confirm - confirm patientUser phoneNumber - unsuccessful 1`] = `Object {}`;
